includes/class-wc-payments-upe-checkout.php

```diff
@@ -161,6 +163,31 @@ public function get_payment_fields_js_config() {
 		return $payment_fields;
 	}
 
+	/**
+	 * This method extracts payment intent data and assumes failed state of operating flow, when the intent already
+	 * bound to an order or intent is processed but order state is not updated yet. Hence, we'll do some checks to
+	 * ensure buyers session stores a usable intent.
+	 *
+	 * Note: I know we should not enter a failed state, but real-live environments behaviour is unpredictable.
+	 *
+	 * @return string|null
+	 */
+	private function get_payment_intent_data_from_session() {
```

Comment:
This part of PR clearly states we are still determining where precisely the root cause.

Could changes in hooks we are bound to ('woocommerce_order_status_changed' now) solve the missing intent cleanup? Possibly, but we need steps to reproduce.

Reply:
+1 for this improvement.